File: problem_set_1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

// Harvard CS50X Problem Set Submission
// Problem Set 1 - Cash
// By Joshua Lehman - April 20, 2024

int calculate_change(int cents, int centsPer);

int main(void)
{
    // Prompt the user to enter the change required, in cents
    int cents;
    int changeCount;
    int centsPerQuarter = 25;
    int centsPerDime = 10;
    int centsPerNickel = 5;
    int centsPerPenny = 1;

    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    // Calculate how many quarters you should give a customer
    int quarters = calculate_change(cents, centsPerQuarter);

    // Subtract the value of those quarters from cents
    cents = cents - (quarters * centsPerQuarter);

    // Calculate how many dimes you should give a customer
    int dimes = calculate_change(cents, centsPerDime);

    // Subtract the value of those dimes from remaining cents
    cents = cents - (dimes * centsPerDime);

    // Calculate how many nickels you should give a customer
    int nickels = calculate_change(cents, centsPerNickel);

    // Subtract the value of those nickels from remaining cents
    cents = cents - (nickels * centsPerNickel);

    // Calculate how many pennies you should give a customer
    int pennies = calculate_change(cents, centsPerPenny);

    // Subtract the value of those pennies from remaining cents
    cents = cents - (pennies * centsPerPenny);

    // Sum the number of quarters, dimes, nickels and pennies
    changeCount = quarters + dimes + nickels + pennies;

    // Print the sum
    printf("%i\n", changeCount);
}

int calculate_change(int cents, int centsPer)
{
    // calculate how many quarters to give to the customer
    int counter = 0;

    while (cents >= centsPer)
    {
        counter++;
        cents = cents - centsPer;
    }
    return counter;
}

File: problem_set_6/sentimental-cash/cash.py
# Harvard CS50X Problem Set Submission
# Problem Set 6 - Sentimental Cash
# By Joshua Lehman - May 7, 2024

from cs50 import get_float


def calculate_change(cents, centsPer):
    counter = 0

    while cents >= centsPer:
        counter += 1
        cents = cents - centsPer
        # print(f"Cents: {cents}")
        # print(f"Counter: {counter}")
    return counter


def main():

    changeCount = 0
    centsPerQuarter = 25
    centsPerDime = 10
    centsPerNickel = 5
    centsPerPenny = 1

    # Prompt user for change owed

    while True:
        cents = get_float("Change owed: ") * 100
        if cents > 0:
            break

    # Calculate how many quarters you should give a customer
    quarters = calculate_change(cents, centsPerQuarter)

    # Calculate how many quarters you should give a customer
    cents = cents - (quarters * centsPerQuarter)

    # Calculate how many quarters you should give a customer
    dimes = calculate_change(cents, centsPerDime)

    # Calculate how many quarters you should give a customer
    cents = cents - (dimes * centsPerDime)

    # Calculate how many quarters you should give a customer
    nickels = calculate_change(cents, centsPerNickel)

    # Calculate how many quarters you should give a customer
    cents = cents - (nickels * centsPerNickel)

    # Calculate how many quarters you should give a customer
    pennies = calculate_change(cents, centsPerPenny)

    # Calculate how many quarters you should give a customer
    cents = cents - (pennies * centsPerPenny)

    # Sum the number of quarters, dimes, nickels and pennies
    changeCount = quarters + nickels + dimes + pennies

    print(f"{changeCount}\n")


main()
